tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -112,29 +113,31 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 	 * Note that only one cluster can run in host network mode at the same time.
 	 */
 	ginkgo.It("Switching back and forth between pod network and host network", func() {
-		// TODO do not skip this if AdvancedStatefulSet feature is enabled
-		serverVersion, err := c.Discovery().ServerVersion()
-		if err != nil {
-			panic(err)
-		}
-		sv := utilversion.MustParseSemantic(serverVersion.GitVersion)
-		klog.Infof("ServerVersion: %v", serverVersion.String())
-		if sv.LessThan(utilversion.MustParseSemantic("v1.13.11")) || // < v1.13.11
-			(sv.AtLeast(utilversion.MustParseSemantic("v1.14.0")) && sv.LessThan(utilversion.MustParseSemantic("v1.14.7"))) || // >= v1.14.0 but < v1.14.7
-			(sv.AtLeast(utilversion.MustParseSemantic("v1.15.0")) && sv.LessThan(utilversion.MustParseSemantic("v1.15.4"))) { // >= v1.15.0 but < v1.15.4
-			// https://github.com/pingcap/tidb-operator/issues/1042#issuecomment-547742565
-			framework.Skipf("Skipping HostNetwork test. Kubernetes %v has a bug that StatefulSet may apply revision incorrectly, HostNetwork cannot work well in this cluster", serverVersion)
+		if !ocfg.Enabled(features.AdvancedStatefulSet) {
+			serverVersion, err := c.Discovery().ServerVersion()
+			framework.ExpectNoError(err, "failed to fetch Kubernetes version")
+			sv := utilversion.MustParseSemantic(serverVersion.GitVersion)
+			klog.Infof("ServerVersion: %v", serverVersion.String())
+			if sv.LessThan(utilversion.MustParseSemantic("v1.13.11")) || // < v1.13.11
+				(sv.AtLeast(utilversion.MustParseSemantic("v1.14.0")) && sv.LessThan(utilversion.MustParseSemantic("v1.14.7"))) || // >= v1.14.0 but < v1.14.7
+				(sv.AtLeast(utilversion.MustParseSemantic("v1.15.0")) && sv.LessThan(utilversion.MustParseSemantic("v1.15.4"))) { // >= v1.15.0 but < v1.15.4
+				// https://github.com/pingcap/tidb-operator/issues/1042#issuecomment-547742565
+				framework.Skipf("Skipping HostNetwork test. Kubernetes %v has a bug that StatefulSet may apply revision incorrectly, HostNetwork cannot work well in this cluster", serverVersion)
+			}
+			ginkgo.By(fmt.Sprintf("Testing HostNetwork feature with Kubernetes %v", serverVersion))
+		} else {
+			ginkgo.By("Testing HostNetwork feature with Advanced StatefulSet")
```

Comment:
Advanced StatefulSet is based on Kubernetes 1.16 stateful set controller, so it does not have this issue